File: grappling_hook/cet/ui_controls/grapple_desiredlength.lua
-- def is models\viewmodels\GrappleDesiredLength
-- style_graphics is models\stylesheet\Graphics
function Draw_GrappleDesiredLength(def, style_graphics, screenOffset_x, screenOffset_y, parent_width, parent_height)
    if not def.should_show then
        do return end
    end

    local color
    if def.isHighlight then
        color = style_graphics.line_color_highlight_abgr
    elseif def.isStandardColor then
        color = style_graphics.line_color_standard_abgr
    else
        color = style_graphics.line_color_gray_abgr
    end

    local center_x = parent_width / 2
    local center_y = parent_height / 2

    local halfHeight = def.height / 2

    --NOTE: 0 is at to, since desired is from the anchor point
    local x = GetScaledValue(def.to_x, def.from_x, 0, 1, def.percent)
    x = center_x + x

    local y = center_y + def.y

    Draw_Line(screenOffset_x, screenOffset_y, x, y - halfHeight, x, y + halfHeight, color, style_graphics.line_thickness_main)
end

File: grappling_hook/models/viewmodels/GrappleDesiredLength.cs
namespace models.viewmodels
{
    public class GrappleDesiredLength
    {

        //TODO: Dashed


        public bool should_show { get; set; }

        /// <summary>
        /// True: Use the standard color
        /// False: Use the gray color
        /// </summary>
        public bool isStandardColor { get; set; }

        /// <summary>
        /// Set to true when they hover over options that will modify these lines
        /// </summary>
        public bool isHighlight { get; set; }

        /// <summary>
        /// This is the height of the line that actually gets drawn (half is above, half is below)
        /// </summary>
        public double height { get; set; }

        /// <summary>
        /// This is where to actually draw the line
        /// NOTE: percent of zero draws at to_x, since desired length is relative to the anchor point
        /// </summary>
        public double percent { get; set; }

        public double from_x { get; set; }
        public double to_x { get; set; }

        public double y { get; set; }
    }
}
